creator/ingest_runs/signals.py

```diff
@@ -9,7 +17,68 @@ def ingest_run_pre_save(sender, instance, *args, **kwargs):
     Set the IngestRun's input_hash from its input parameters and the name
     using the version ids if they exist
     """
-    # Populate the input_hash and name from the version ids
+    # Populate the input_hash and name from the version IDs
     if instance.versions.count() > 0:
         instance.name = instance.compute_name()
         instance.input_hash = instance.compute_input_hash()
+
+
+@receiver(pre_delete, sender=Version)
+def version_pre_delete(sender, instance, using, *args, **kwargs):
+    """
+    When a Version is deleted, cancel any IngestRuns
+    that include this Version.
+    """
+    logging.info(
+        f"Canceling all ingest runs containing version {instance.pk} before"
+        f" its deletion."
+    )
+    cancel_invalid_ingest_runs(instance)
+
+
+def cancel_invalid_ingest_runs(version):
```

Comment:
I don't see this method being used anywhere else? I think this functionality could just be called directly on line 36

Reply:
it's used in the signals tests